### PR TITLE
Fix font not resetting when logging out

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/17228

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix font not resetting when logging out ([\#8670](https://github.com/matrix-org/matrix-react-sdk/pull/8670)). Fixes vector-im/element-web#17228.

### CODE REVIEW

File: src/settings/watchers/FontWatcher.ts [hunk 1]
```diff
@@ -34,27 +37,41 @@ export class FontWatcher implements IWatcher {
...
+        });
+    }
+
+    private onAction = (payload: ActionPayload) => {
         if (payload.action === Action.UpdateFontSize) {
             this.setRootFontSize(payload.size);
         } else if (payload.action === Action.UpdateSystemFont) {
-            this.setSystemFont(payload);
+            this.setSystemFont(payload as UpdateSystemFontPayload);
+        } else if (payload.action === "on_logged_out") {
```
Comment (andybalaam) [inline src/settings/watchers/FontWatcher.ts:61]: Should we add an `Action.LogOut` and `Action.LogIn` ? Why is `"on_logged_out"` a plain string?

File: src/settings/watchers/FontWatcher.ts [hunk 2]
```diff
@@ -20,9 +20,12 @@ import IWatcher from "./Watcher";
 import { toPx } from '../../utils/units';
 import { Action } from '../../dispatcher/actions';
 import { SettingLevel } from "../SettingLevel";
+import { UpdateSystemFontPayload } from "../../dispatcher/payloads/UpdateSystemFontPayload";
+import { ActionPayload } from "../../dispatcher/payloads";
 
 export class FontWatcher implements IWatcher {
```
Comment (andybalaam) [inline src/settings/watchers/FontWatcher.ts:26]: Some tests for the new functionality on this class might be possible. Desirable if possible.

**Review by andybalaam — CHANGES_REQUESTED** [review]

File: src/settings/watchers/FontWatcher.ts (hunk 1 above)
Comment (t3chguy) [reply src/settings/watchers/FontWatcher.ts:61]: These just haven't been brought over into the typed world yet, happy to do it as part of the PR

Comment (andybalaam) [reply src/settings/watchers/FontWatcher.ts:61]: Not insisting, but would be nice if possible.

Comment (t3chguy) [reply src/settings/watchers/FontWatcher.ts:61]: Happy to, we need to ditch the old string ones sooner or later, they make the dispatcher not type-safe

**Review by andybalaam — APPROVED** [review]
Awesome, thanks!

**Pull request merged**